hs_composite_resource/models.py

```diff
@@ -65,17 +66,76 @@ def can_be_published(self):
             # url file cannot be published
             if 'url' in lf.extra_data:
                 return False
+            # check for model instance linked to model program
```

Comment:
I think this logic belongs with the model instance logical file, not the composite resource.  I suggest adding a property to the abstractlogicalfile class and put lines 64-68 in there.  Then overwrite the property in the model instance logical file and place all of the logic below into there.  

Then just loop through all logical files and call the property on each logical file until you hit a False return value.